packages/graphql/src/services/graphql.service.ts

```diff
@@ -141,13 +145,14 @@ export class GraphqlService implements BackendService {
 
     // only add pagination if it's enabled in the grid options
     if (this._gridOptions.enablePagination !== false) {
-      datasetFilters = {
-        ...this.options.paginationOptions,
-        first: ((this.options.paginationOptions && this.options.paginationOptions.first) ? this.options.paginationOptions.first : ((this.pagination && this.pagination.pageSize) ? this.pagination.pageSize : null)) || this.defaultPaginationOptions.first
-      };
+      datasetFilters = {};
 
-      if (!this.options.isWithCursor) {
+      if (this.options.isWithCursor && this.options.paginationOptions) {
+        datasetFilters = { ...this.options.paginationOptions };
+      }
+      else {
         const paginationOptions = this.options?.paginationOptions;
+        datasetFilters.first = ((this.options.paginationOptions && this.options.paginationOptions.first) ? this.options.paginationOptions.first : ((this.pagination && this.pagination.pageSize) ? this.pagination.pageSize : null)) || this.defaultPaginationOptions.first;
```

Comment:
I think this could be simplified with optional chaining `?.`
```ts
this.options?.paginationOptions.first ? condition1 : condition2;
```

Reply:
Made the reading even easier by using null coalescing `??`